File: celladapter/src/main/java/io/techery/celladapter/select/SelectableCell.java
package io.techery.celladapter.select;

import android.view.View;

import io.techery.celladapter.Cell;
import io.techery.celladapter.select.mode.SelectionManager;

public abstract class SelectableCell<ITEM, LISTENER extends Cell.Listener<ITEM>> extends Cell<ITEM, LISTENER> {

    protected SelectionManager selectionManager;

    public SelectableCell(View view) {
        super(view);
    }

    public void setSelectionManager(SelectionManager selectionManager) {
        this.selectionManager = selectionManager;
    }
}

File: celladapter-kotlin/src/main/kotlin/io/erva/celladapter/x/select/SelectableCell.kt
package io.erva.celladapter.x.select

import android.view.View

import io.erva.celladapter.x.Cell
import io.erva.celladapter.x.select.mode.SelectionManager

abstract class SelectableCell<ITEM : Any, out LISTENER : Cell.Listener<ITEM>>(view: View) : Cell<ITEM, LISTENER>(view) {

    lateinit var selectionManager: SelectionManager
}